code/optimize/src/test/java/com/adobe/marketing/mobile/optimize/OptimizeExtensionTests.java

```diff
@@ -1200,6 +1203,56 @@ public void testHandleEdgeResponse_requestEventIdNotBeingTracked() throws Except
         }
     }
 
+    @Test
+    public void testHandleEdgeResponse_missingItemData() throws IOException {
+        // Setup
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, Object> edgeResponseData =
+                objectMapper.readValue(
+                        Objects.requireNonNull(getClass().getClassLoader())
+                                .getResource(
+                                        "json/EVENT_DATA_EDGE_RESPONSE_MISSING_ITEM_DATA.json"),
+                        HashMap.class);
+
+        Event testEvent =
+                new Event.Builder(
+                                "AEP Response Event Handle",
+                                "com.adobe.eventType.edge",
+                                "personalization:decisions")
+                        .setEventData(edgeResponseData)
+                        .build();
+
+        ArgumentCaptor<Event> eventCaptor = ArgumentCaptor.forClass(Event.class);
+        extension.setUpdateRequestEventIdsInProgress(
+                "AAAAAAAA-AAAA-AAAA-AAAA-AAAAAAAAAAAA",
+                new ArrayList<DecisionScope>() {
+                    {
+                        add(new DecisionScope("someDecisionScope"));
+                    }
+                });
+        extension.handleEdgeResponse(testEvent);
+        Mockito.verify(mockExtensionApi, Mockito.times(1)).dispatch(eventCaptor.capture());
+
+        List<Map<String, Object>> propositionsList =
+                (List<Map<String, Object>>)
+                        eventCaptor.getValue().getEventData().get("propositions");
+
+        Assert.assertNotNull("Propositions list should not be null", propositionsList);
+        try {
+            List<Map<String, Object>> items =
```

Comment:
Could you please follow existing test conventions, convert the proposition map to `OptimizePropositions` using `OptimizeProposition.fromEventData()` and do `optimizeProposition.getOffers()`? Please refer `testHandleEdgeResponse_validProposition` for more details

Reply:
Updated @spoorthipujariadobe